### DISCUSSION TITLE
basic-extra-meal/generic

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#246] [Announcements]
# basic-extra-meal/generic

本文详细讲解了 Java 泛型的概念、原理及应用技巧，为您展示了如何通过泛型提高代码的可重用性、类型安全和可读性。学习本文将帮助您更好地掌握 Java 泛型编程，提高编程效率与质量。

https://javabetter.cn/basic-extra-meal/generic.html

**Comment by Eivae** [comment]
不是无法从<? super T>这样的类型中提取数据吗，怎么最后这个例子又可以了？

**Reply by xmerge** [reply]
通配符分为“无界通配符”、“上限通配符”、“下限通配符”。其中“无界通配符”和“上限通配符”只能读取和添加“null”对象，而“下限通配符”可以读取和写入

**Comment by Knightzh** [comment]
<? super T>为啥子前后矛盾，泛型通配符最后说不能get，结尾代码又get了，到底是啥意思

**Reply by xmerge** [reply]
通配符分为“无界通配符”、“上限通配符”、“下限通配符”。其中“无界通配符”和“上限通配符”只能读取和添加“null”对象，而“下限通配符”可以读取和写入
